src/gamestates/TestState.cpp

```diff
@@ -16,25 +16,34 @@ TestState::TestState(SDL_Renderer& renderer)
 }
 
 gamestates::GameStateName TestState::run() {
-    Keyboard keyboard = Keyboard();
+    // Initialize engine critical components
     GraphicsHandler graphics_handler(renderer_);
-    graphics_handler.init();
+    Universe universe(graphics_handler);
     std::unique_ptr<Events> events(new Events());
     std::unique_ptr<I_CollisionDetector> collision_detector(new CollisionDetector());
-    Universe universe = Universe(graphics_handler);
     GameObjectMediator game_object_mediator(universe, *collision_detector);
+    Keyboard keyboard;
 
+    // Setup engine critical components
```

Comment:
I'm thinking we can bake this into a gamestate. Like does a gamestate really need that much control of whether a universe and etc are created or not? I don't think so, they can just have access functions.